#870 python http template (#881)
Azure/azure-functions-templates: Functions.Templates/Templates/HttpTrigger-Python/README.md

# Python HttpTrigger

> The Azure Functions Python HTTP Trigger creates an endpoint: `<function-host>/api/<function-name>`. When an HTTP request is made against that endpoint, the `main()` function in `__init__.py` is called.

Reviewer: ```suggestion
The HTTP trigger lets you invoke a function with an HTTP request. You can use an HTTP trigger to build serverless APIs and respond to webhooks. It creates an endpoint like  `https://<app-name>.azurewebsites.net/api/api/<function-name>`. When an HTTP request is made against that endpoint, the `main()` function in `__init__.py` is called.
```

Author: Accepting your suggestion by changing my own code. Just wanted to keep the commit tree clean.

> Throughout this documentation variables in angle brackets `<>` are placeholders and will be different in your application.

> ## Default Structure

Author: Reminder to self- rename to Template Structure

Reviewer: I think "Default" is fine.

``` text
<function-name>/      # directory holding all files for the HttpTrigger Function
└── __init__.py     # this module will be called when the Function is run
    function.json   # defines that __init__.py will be called and which bindings are used

>     sample.dat      # contains a sample input for the Function

Reviewer: You could also leave it out since it's not integral to the working of the function. Up to you.

>     sample.dat      # contains a sample input for the Function

Author: I added it just because I have a programmer's penchant for lists that include themselves.

```

> ## Testing the Template Function

Reviewer: ```suggestion
## Run the function locally
```

The default function template when a new HttpTrigger is created takes a value for `name` and returns `Hello <name>!`. To test this functionality, from the root of the Azure Function App, execute

``` bash
func host start
```

> 

Reviewer: ```suggestion
If you're running in VS Code, you can simply press F5 to test your function locally. 
```

Once running, you should see

``` text
Http Functions:

        <function-name>: [GET,POST] http://localhost:7071/api/<function-name>
```

Open the URL in a browser. You should get the response

``` text
Please pass a name on the query string or in the request body
```

To the URL, append `?name=Sophia` so it reads

``` text
http://localhost:7071/api/<function-name>?name=Sophia
```

You should now get the response

``` text
Hello Sophia!
```

> ## Useful Resources

Reviewer: ```suggestion
## Resources
```

> - Software: [Postman](https://www.getpostman.com/) is a cross-platform application for testing HTTP requests.

Reviewer: This can be tricky wrt referencing third-party resources. Let's hold back for now. I'll follow up and confirm if we can add this to our templates. 

```suggestion

```

> - Guide: [Create an HTTP triggered function in Azure](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-python)  

Reviewer: ```suggestion
- [Azure Functions Python developer reference](https://aka.ms/python-functions-guide)  
```

> - Examples: [Serverless Library - Python](https://serverlesslibrary.net/?technology=Functions%202.x&language=Python)

Reviewer: ```suggestion
- [HTTP triggers and bindings](https://docs.microsoft.com/azure/azure-functions/functions-bindings-http-webhook)
- [Serverless Library](https://aka.ms/python-functions-samples)
```
